hubbard/hamiltonian.py

```diff
@@ -630,6 +632,9 @@ def converge(self, calc_n_method, tol=1e-6, mixer=None, steps=100, fn=None, prin
                     print('   %i iterations completed:' % i, dn, self.Etot)
                 if fn:
                     self.write_density(fn, 'a')
+            if max_iter is not None:
```

Comment:
Sorry to chip in, instead of doing
```python
if max_iter is not None:
...
```
you could move the check outside, and do:
```python
if max_iter is None:
    max_iter = -1
while ...

    if i == max_iter: ..
```
this would still work and no extra check is necessary.

Reply:
thanks!